### ISSUE TITLE
Improve error reporting for products images import

### ISSUE THREAD

**Issue by leoquijano** [magento/magento2#4711]
When importing products using the System -> Import Products feature, if there problems importing the images, Magento just fails with:

```
1. Imported resource (image) could not be downloaded from external resource due to timeout or access permissions in rows: ...
```

No information is given about the cause of the problem, nor any help for the developer to sort it out. Generic calls to fix permissions don't really help, and it makes the process of importing products way longer than expected.

As explained [here](https://mage2.pro/t/topic/1099), exception handling could be improved to provide helpful messages to the developer. In particular, the following can be added to `vendor/magento/module-catalog-import-export/Model/Import/Product.php`:

``` php
protected function uploadMediaFiles($fileName, $renameFileOff = false)
{
    try {
        $res = $this->_getUploader()->move($fileName, $renameFileOff);
        return $res['file'];
    } catch (\Exception $e) {
        return '';
    }
}
```

Swallowing exceptions is a very bad thing, so it'd be a good idea to improve it like this:

``` php
protected function uploadMediaFiles($fileName, $renameFileOff = false)
{
    try {
        $res = $this->_getUploader()->move($fileName, $renameFileOff);
        return $res['file'];
    } catch (\Exception $e) {
        $this->_logger->critical($e);
        return '';
    }
}
```

**Comment by wert2all** [comment]
Please, provide the used version. If the problem is actual for a specific branch, please, specify it and be sure that the latest update was used.

**Comment by leoquijano** [comment]
Thanks @wert2all. This is branch 2.0, specifically in my case, version 2.0.7:

https://github.com/magento/magento2/blob/2.0/app/code/Magento/CatalogImportExport/Model/Import/Product.php

**Comment by BaDos** [comment]
Hi @leoquijano 

Thank you for reporting.
We've created the internal ticket MAGETWO-58375 to resolve this issue.

**Comment by magento-engcom-team** [comment]
@leoquijano, thank you for your report.
We've created internal ticket(s) MAGETWO-58375 to track progress on the issue.

**Comment by leoquijano** [comment]
Just fyi, it's present in v2.2.0 too:

https://github.com/magento/magento2/blob/2.2/app/code/Magento/CatalogImportExport/Model/Import/Product.php

**Comment by okorshenko** [comment]
@leoquijano the issue has been fixed and delivered to 2.2-develop branch. Will be available with 2.2.2 release

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-82886

**Comment by leoquijano** [comment]
Thank you.

**Comment by ishakhsuvarov** [comment]
Hi @leoquijano. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1201 by @nmalevanec in 2.3-develop branch
Related commit(s):
 - 0d01f54f07794cbc8b2974a9aebb99ebecf4ea12

The fix will be available with the upcoming patch release.
